core/src/main/java/com/graphhopper/routing/profiles/EnumEncodedValue.java

```diff
@@ -36,6 +36,10 @@ public EnumEncodedValue(String name, Class<E> enumType, boolean storeTwoDirectio
         arr = enumType.getEnumConstants();
     }
 
+    public E[] getValues() {
```

Comment:
Look all I had to do was add this method and I got rid of `EncodedValueFactory#findValues` and we no longer have to pass the factory everywhere.

Reply:
Pretty! And solves the problem that it had to stay in-sync too. Sometimes I see the wald vor lauter bäumen nicht :)